packages/dht/src/connection/websocket/WebsocketConnector.ts

```diff
@@ -308,7 +315,9 @@ export class WebsocketConnector {
                 targetPeerDescriptor
             )
             managedConnection.setRemotePeerDescriptor(sourcePeerDescriptor)
-            if (targetPeerDescriptor && !areEqualPeerDescriptors(this.localPeerDescriptor!, targetPeerDescriptor)) {
+            if (!isCompatibleVersion(sourceVersion, version)) {
+                managedConnection.rejectHandshake(HandshakeError.UNSUPPORTED_VERSION)
```

Comment:
Duplication of 
```ts
if (!isCompatibleVersion(sourceVersion, version)) {
    managedConnection.rejectHandshake(HandshakeError.UNSUPPORTED_VERSION)
}
```
Move before (outside) if-else block?

Reply:
Yeah would be good to not have to do duplication here but it is necessary in this case